warehouse/router_identities.go

```diff
@@ -152,16 +149,16 @@ func (wh *HandleT) hasLocalIdentityData(warehouse model.Warehouse) (exists bool)
 `,
 		warehouseutils.IdentityMergeRulesTableName(warehouse),
 	)
-	err := wh.dbHandle.QueryRow(sqlStatement).Scan(&exists)
+	err := r.dbHandle.QueryRow(sqlStatement).Scan(&exists)
 	if err != nil {
 		// TODO: Handle this
 		panic(fmt.Errorf("Query: %s\nfailed with Error : %w", sqlStatement, err))
 	}
 	return
 }
 
-func (wh *HandleT) hasWarehouseData(ctx context.Context, warehouse model.Warehouse) (bool, error) {
-	whManager, err := manager.New(wh.destType, wh.conf, wh.Logger, wh.stats)
+func (r *Router) hasWarehouseData(ctx context.Context, warehouse model.Warehouse) (bool, error) {
+	whManager, err := manager.New(r.destType, r.conf, r.logger, r.stats)
```

Comment:
We might not need to create a new instance each time.

Reply:
This is used in other places too so it might make sense to create it only once (see line 399).